### PR TITLE
[WFCORE-6351] The publish-configuration command doesn't throw an error when the git location is invalid

### PR DESCRIPTION
**Pull request by ivassile**
Issue: https://issues.redhat.com/browse/WFCORE-6351

### CODE REVIEW

File: server/src/main/java/org/jboss/as/server/controller/git/GitConfigurationPersister.java [hunk 1]
```diff
@@ -154,6 +155,8 @@ public String publish(String name) throws ConfigurationPersistenceException {
             } catch (GitAPIException ex) {
                 throw MGMT_OP_LOGGER.failedToPublishConfiguration(ex, name, ex.getMessage());
             }
+        } else {
+            throw MGMT_OP_LOGGER.failedToPublishConfiguration(null, name, INVALID_REMOTE_NAME);
```
Comment (ehsavoie) [inline server/src/main/java/org/jboss/as/server/controller/git/GitConfigurationPersister.java:159]: This should be properly managed in the logger

**Review by ehsavoie — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/jboss/as/server/controller/git/GitConfigurationPersister.java (hunk 1 above)
Comment (ivassile) [reply server/src/main/java/org/jboss/as/server/controller/git/GitConfigurationPersister.java]: Changed the implementation.

**Review by yersan — CHANGES_REQUESTED** [review]
Hi @ivassile , this needs some adjustments in the test cases; see GitRepositoryTestCase.historyAndManagementOperationsTest

**Comment by ivassile** [comment]
@yersan Discovered that we have another problem which causes the test failure and created new issue WFCORE-6358 [1]. I'm working on the issue and after WFCORE-6358 is fixed, this PR will pass. 

[1] https://issues.redhat.com/browse/WFCORE-6358

**Comment by yersan** [comment]
Ok, @ivassile , you can combine both on the same PR with two commits, getting the new PR superseding this one; up to you.

**Comment by yersan** [comment]
@ivassile Any chances to get this one and [WFCORE-6358](https://issues.redhat.com/browse/WFCORE-6358) in place soon? CC @ehsavoie

**Comment by ivassile** [comment]
> @ivassile Any chances to get this one and [WFCORE-6358](https://issues.redhat.com/browse/WFCORE-6358) in place soon? CC @ehsavoie

@yersan I was finishing up some Elytron RFEs for EAP 8. I'll work on the issues today. Thanks!

**Comment by ivassile** [comment]
@yersan @ehsavoie After debugging the issue, I realized that the problem is actually the failing GitRepositoryTestCase.historyAndManagementOperationsTest. This test was never worked before but we catch the exception now because of this change. I removed the case ":publish-configuration => push to origin" because we don't have "origin" (default) remote in local. This case is applicable to RemoteGitRepositoryTestCase. We don't have new issue WFCORE-6358 and it's now closed.

**Comment by yersan** [comment]
Hi @ivassile sadly this one got a conflict, could you rebase fixing the conflict?

**Comment by ivassile** [comment]
> Hi @ivassile sadly this one got a conflict, could you rebase fixing the conflict?
Rebased on latest changes.

**Review by yersan — APPROVED** [review]
The test failure is unrelated

**Comment by yersan** [comment]
Thanks @ivassile

**Pull request merged**